validation-api/azure_utils/auth.py

```diff
@@ -15,29 +15,24 @@ def read_config(config_path):
     return config
 
 
-def obtain_credential(config, credential_type="default"):
-    """Obtains client credentials from Azure KeyVault.
-    Args:
-        config (dict): Dictionary of configuration values.
+def obtain_sp_credential():
+    """Obtains service principal credentials from Azure.
     Returns:
-        Instance of ClientSecretCredential.
+        Instance of DefaultAzureCredential.
     """
 
     # Since this will be run from a Docker container,
-    # we only use the SP Credential to Authenticate.
+    # we only use the SP Credential to authenticate.
 
-    # Pull values from injected environment variables
-    tenant_id = os.environ.get("TENANT_ID", "")
-    application_id = os.environ.get("APPLICATION_ID", "")
-    client_secret = os.environ.get("SP_SECRET", "")
+    # Check that env variables are set
+    tenant_id = os.environ.get("AZURE_TENANT_ID", "")
+    application_id = os.environ.get("AZURE_CLIENT_ID", "")
+    client_secret = os.environ.get("AZURE_CLIENT_SECRET", "")
     if "" in [tenant_id, client_secret, application_id]:
         error_message = "Azure secrets not found in environment. Ensure secrets are configured properly."
         raise ValueError(error_message)
 
-    sp_credential = ClientSecretCredential(
-        tenant_id=tenant_id,
-        client_id=application_id,
-        client_secret=client_secret,
-    )
+    # The DefaultAzureCredential reads from the environment directly
```

Comment:
So we check that the env vars are there, and then let `DefaultAzureCredential` pull them?

Reply:
Yep!